content/docs/iac/concepts/resources/functions.md

```diff
@@ -225,40 +227,54 @@ resources:
+{{% / notes %}}
+
+## Direct form and output form
+
 Provider functions are exposed in each language as regular functions, in two variations:
 
- 1. The **direct form** accepts plain arguments (e.g. `string`, as opposed to `pulumi.Input<string>`) and returns an asynchronous value (e.g. a `Promise` in NodeJS, or a `Task` in Python), or blocks until the result is available. These functions are typically named, e.g., `getFoo()`.
- 2. The **output form** accepts `Input` values (or plain values) and returns an [Output](/docs/concepts/inputs-outputs/). These functions are typically named, e.g., `getFooOutput()`.
+ 1. The **direct form** accepts plain arguments (e.g. `string`, as opposed to `pulumi.Input<string>`) and returns an asynchronous value or blocks until the result is available.
+ 1. The **output form** accepts Pulumi Inputs as arguments and returns a Pulumi as a result. For more information on these types, see [Inputs and Outputs](/docs/concepts/inputs-outputs/).
 
 The [Pulumi Registry](/registry) contains authoritative documentation for all provider functions.
 
-#### Invoke options
-
-Functions also accept "invoke options", similar to the way Pulumi resources accept [resource options](/docs/concepts/options/). Invoke options may be specified either as an object or as a list of arguments depending on the language you're writing your Pulumi program in. The options are as follows:
+## Invoke options
 
-- `dependsOn`: An array of resources that this function depends on, see [Dependencies and ordering](#dependencies-and-ordering). This option is only available on Output form invocations.
+In addition to function arguments, provider functions also accept "invoke options", similar to the way Pulumi resources accept [resource options](/docs/concepts/options/). Invoke options may be specified either as an object or as a list of arguments depending on the language you're writing your Pulumi program in. The available options are:
 
-- `parent`: Supply a parent resource for this function call. Much like the [parent resource option](/docs/concepts/options/parent/), the parent will be consulted when determining the provider to use.
+* `dependsOn`: An array of resources that this function depends on. This option is only available in the Output form of a provider function. See [Dependencies and ordering](#dependencies-and-ordering) for a full explanation.
+* `parent`: Supply a parent resource for this function call. Much like the [parent resource option](/docs/concepts/options/parent/), the parent will be consulted when determining the provider to use.
+* `pluginDownloadURL`: Pass a URL from which the provider plugin should be fetched. This may be necessary for third-party packages such as those not hosted at [https://get.pulumi.com](https://get.pulumi.com).
```

Comment:
`pluginDownloadURL` and `version` should not be explicitly passed by users. They are automatically pre-set by the calling SDK. 

You can see this looking at 3rd party SDKs: https://github.com/wttech/pulumi-aem/blob/f1123dcace8f7812a0239f22d2a12d3b299d2b68/sdk/go/aem/internal/pulumiUtilities.go#L178-L182.